src/Microsoft.Health.Fhir.SqlServer/Features/Watchdogs/DefragWatchdog.cs

```diff
@@ -145,19 +148,20 @@ private async Task ExecDefragWithHeartbeatAsync(CancellationToken cancellationTo
                 }
 
                 using IScoped<SqlQueueClient> scopedQueueClient = _sqlQueueClient.Invoke();
-                await scopedQueueClient.Value.ExecuteJobWithHeartbeatAsync(
+                await scopedQueueClient.Value.ExecuteJobWithHeartbeats(
                     QueueType,
                     jobId,
                     job.version,
                     async cancellationSource =>
                     {
                         var split = job.definition.Split(";");
-                        await DefragAsync(split[0], split[1], int.Parse(split[2]), byte.Parse(split[3]) == 1, cancellationSource);
+                        await DefragAsync(split[0], split[1], int.Parse(split[2]), byte.Parse(split[3]) == 1, cancellationSource.Token);
+                        return null;
```

Comment:
Recommend returning Task.CompletedTask. If you return null you will most likely get a null reference exception

Reply:
Defrag test passes... I will follow your recommendation anyway...